### ISSUE TITLE
Move e2e-runner.sh logic into hack/e2e.go

### ISSUE THREAD

**Issue by fejta** [kubernetes/test-infra#1475]
Looks like e2e-runner.sh does the following things, all of which we can migrate into e2e.go:

1) ensure project is set
2) download sh2ju.sh
3) Create artifacts dir
4) Some some default env args
5) Determine if it is running in docker (barf)
6) maybe download gcloud
7) activate the service account
8) maybe setup gci vars
9) print env variables
10) maybe call get kube
11) maybe config-ssh
12) maybe call get kube for a skewed version
13) cd into kubernetes
14) maybe set the priority path (???)
15) set a bunch of e2e go args
16) do testing
17) maybe publish on success
18) maybe upload kube config

**Comment by zmerlynn** [comment]
> maybe set the priority path (???)

Added in #1442 because `e2e-runner.sh` overrides the PATH to the cloud SDK path, and I wanted to ensure we picked up the right kubectl regardless. It can easily be removed if `kops-e2e-runner.sh` is also folded in, but when I created *that* file, you were against adding a lot of this to e2e.go (e.g. I originally coded things so that e2e.go owuld download the `kops` binaries, etc.).

Have you changed your mind? This is an awful lot to add to e2e.go.
